hw3/src/PieChart.js

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import Chart from "chart.js/auto";
+
+const PieChart = () => {
+  const [languages, setLanguages] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const fetchLanguages = async () => {
+      try {
+        const response = await axios.get(
+          "https://cs464p564-frontend-api.vercel.app/api/countries"
+        );
+        const southAmericanCountries = response.data;
+        const allLanguages = southAmericanCountries.flatMap(
+          (country) => country.official_languages
+        );
+        const languageCounts = countLanguages(allLanguages);
+        setLanguages(languageCounts);
+        setLoading(false);
+      } catch (error) {
+        console.error(error);
+        setError(error);
+        setLoading(false);
+      }
+    };
+
+    fetchLanguages();
+  }, []);
+
+  const countLanguages = (languages) => {
+    return languages.reduce((acc, language) => {
+      acc[language] = (acc[language] || 0) + 1;
+      return acc;
+    }, {});
+  };
+
+  useEffect(() => {
+    if (!loading && !error) {
+      const ctx = document.getElementById("chart");
+      new Chart(ctx, {
+        type: "pie",
+        data: {
+          labels: Object.keys(languages),
+          datasets: [
+            {
+              data: Object.values(languages),
+              backgroundColor: [
+                "rgba(255, 99, 132, 0.6)",
```

Comment:
Consider adding more diverse colors as it made some countries hard to tell apart